frontend/javascripts/maintenance_banner.tsx

```diff
@@ -4,24 +4,41 @@ import {
 } from "admin/admin_rest_api";
 import { Alert } from "antd";
 import FormattedDate from "components/formatted_date";
-import { useInterval } from "libs/react_helpers";
+import { useFetch, useInterval } from "libs/react_helpers";
+import { sleep } from "libs/utils";
 import { navbarHeight } from "navbar";
 import { setActiveUserAction } from "oxalis/model/actions/user_actions";
 import { Store } from "oxalis/singletons";
+import { OxalisState } from "oxalis/store";
 import React, { useState } from "react";
+import { useSelector } from "react-redux";
 import { MaintenanceInfo } from "types/api_flow_types";
 
+const INITIAL_DELAY = 5000;
 const INTERVAL_TO_FETCH_MAINTENANCES_MS = 60000;
 
 export function MaintenanceBanner() {
-  const { activeUser, uiInformation } = Store.getState();
+  const activeUser = useSelector((state: OxalisState) => state.activeUser);
```

Comment:
@dieknolle3333 just fyi, I fixed the store access here. When rendering react components, it's always important to properly hook up with the store so that the component updates when the store updates. This listener can be set up with `connect` (from `react-redux`) or in case of functional components `useSelector` (which is better in my opinion since it's less verbose). Your access to `Store.getState()` is fine in event handlers (e.g., onClick handler in a button) when one doesn't need the synchronization. However, during rendering this would mean that the component gets outdated at some point.

Reply:
oh, I understand!